lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```diff
@@ -739,18 +746,22 @@ selectTxs txids = do
 selectTxHistory
     :: forall t. PersistTx t
     => W.WalletId
+    -> W.SortDirection
     -> [Filter TxMeta]
     -> SqlPersistT IO [(W.Hash "Tx", (W.Tx t, W.TxMeta))]
-selectTxHistory wid conditions = do
+selectTxHistory wid dir conditions = do
     -- Note: there are sorted indices on these columns.
     -- The secondary sort by TxId is to make the ordering stable
     -- so that testing with random data always works.
-    let sortOpt = [ Desc TxMetaSlotId, Asc TxMetaTxId ]
     metas <- fmap entityVal <$> selectList
         ((TxMetaWalletId ==. wid) : conditions) sortOpt
     let txids = map txMetaTxId metas
     (ins, outs) <- selectTxs txids
     pure $ txHistoryFromEntity @t metas ins outs
+  where
+    sortOpt = case dir of
+        W.Ascending -> [Asc TxMetaSlotId, Desc TxMetaTxId]
+        W.Descending -> [Desc TxMetaSlotId, Asc TxMetaTxId]
```

Comment:
I think that would break `readTxHistory wid Descending noFilter === reverse (readTxHistory wid Ascending noFilter)` which I thought would be a nice property to have.

🤔

Reply:
Better move the comment about indexes down to the `where` clauses.
With sqlite, for sort speed, it doesn't matter which direction the index is created in, there just needs to be a sorted index.

Reply:
Moved.